pkg/controller/llmisvc/scheduler.go

```diff
@@ -283,13 +288,86 @@ func (r *LLMInferenceServiceReconciler) expectedSchedulerDeployment(ctx context.
 
 	if llmSvc.Spec.Router != nil && llmSvc.Spec.Router.Scheduler != nil && llmSvc.Spec.Router.Scheduler.Template != nil {
 		d.Spec.Template.Spec = *llmSvc.Spec.Router.Scheduler.Template.DeepCopy()
+		for i := range d.Spec.Template.Spec.Containers {
+			if d.Spec.Template.Spec.Containers[i].Name != "main" {
+				continue
+			}
+
+			if slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "--configText") ||
+				slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "-configText") ||
+				slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "--configFile") ||
+				slices.Contains(d.Spec.Template.Spec.Containers[i].Args, "-configFile") {
+				// When the configuration is overridden, don't add/override it.
+				break
+			}
+
+			d.Spec.Template.Spec.Containers[i].Args = append(d.Spec.Template.Spec.Containers[i].Args,
+				"--configText",
+				schedulerConfigText(llmSvc),
+			)
+		}
 	}
 
 	log.FromContext(ctx).V(2).Info("Expected router scheduler deployment", "deployment", d)
 
 	return d
 }
 
+func schedulerConfigText(llmSvc *v1alpha1.LLMInferenceService) string {
```

Comment:
Would it make sense to move it `scheduler_config.go` to keep these things together?